dataverse:unstable increased the version to 6.3

### Prerequisites

* [x] Are you running the expected version of pyDataverse? (check via `pip freeze`).

main branch

### Bug report

The :unstable docker image now ships dataverse 6.3, but DV_VERSION in the docker-compose-test-all.yml is pinned to 6.2, causing several tests to fail.

**1. Describe your environment**

* [x] OS: Ubuntu (I think... in the container)
* [x] pyDataverse: main branch
* [x] Python: 3.11 (in the container)
* [x] Dataverse: 6.3 (!)

**2. Actual behaviour:**

Some tests fail (here one example):
```
_______________________ TestApiToken.test_token_missing ________________________

self = <tests.api.test_api.TestApiToken object at 0xffff8b262fd0>

    def test_token_missing(self):
        BASE_URL = os.getenv("BASE_URL").rstrip("/")
        api = NativeApi(BASE_URL)
        resp = api.get_info_version()
>       assert resp.json()["data"]["version"] == os.getenv("DV_VERSION")
E       AssertionError: assert '6.3' == '6.2'
E         
E         - 6.2
E         + 6.3

tests/api/test_api.py:75: AssertionError
```

**3. Expected behaviour:**

The tests pass.

**4. Steps to reproduce**

1. `sh run-tests.sh`.

❌ Unit tests failed. Printing logs...

**5. Possible solution**

- Update DV_VERSION in docker-compose-test-all.yml to 6.3.
- Do not use :unstable but pin to a specific image version. Since Dataverse only tags two versions, I guess a hash might work instead.
- Ask if Dataverse's docker images can be tagged with the versions, so that we could even test against different versions easily, for example to handle different API changes.

I will provide a PR to increase the version to 6.3, as described in the first variant. I don't think the second is feasible right now, but I think it might be worth exploring the third option, though I really don't know what the status of the Dataverse development in general is on that matter.

--- a/docker/docker-compose-test-all.yml
+++ b/docker/docker-compose-test-all.yml
@@ -5,7 +5,7 @@ services:
     image: python:${PYTHON_VERSION}-slim
     environment:
       BASE_URL: http://dataverse:8080
-      DV_VERSION: 6.2
+      DV_VERSION: 6.3
     networks:
       - dataverse
     volumes:

